docs/content/development/hacking-on-gitea.en-us.md

```diff
@@ -191,6 +191,22 @@ Before committing, make sure the linters pass:
 make lint-frontend
 ```
 
+## Typescript
+
+Gitea is in the process of migrating to type-safe Typescript. Here are some specific guidelines regarding Typescript in the codebase:
+
+### Use type aliases instead of interfaces
+
+Prefer to use type aliases because they can represent any type and are generally more flexible to use than interfaces.
+
+### Use separate type imports
+
+We use `verbatimModuleSyntax` so type imports and non-type imports from the same file must be split into two `import type` statements. This enables the typescript compiler to completely eliminate the type import statements during compilation.
+
+### Use `@ts-expect-error` instead of `@ts-ignore`
+
+Both annotations should be avoided, but if you have to use them, use `@ts-expect-error` because it will not leave ineffective statements after the issue is fixed.
+
```

Comment:
I think these sections should be in frontend guideline doc (and that doc needs to be updated)

Reply:
Ah, I forgot we had a separate document for it.